transport/httptransport/libp2p_server.go

```diff
@@ -0,0 +1,591 @@
+package httptransport
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/filecoin-project/boost/car"
+	"github.com/filecoin-project/boost/transport/types"
+	blockstore "github.com/ipfs/go-ipfs-blockstore"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/peer"
+	gostream "github.com/libp2p/go-libp2p-gostream"
+	"github.com/multiformats/go-multiaddr"
+	"golang.org/x/xerrors"
+)
+
+// Libp2pCarServer serves deal data by matching an auth token to the root CID
+// of a DAG in a blockstore, and serving the data as a CAR
+type Libp2pCarServer struct {
```

Comment:
Please can we add some more Info logging for the Transfer lifecycle /ability to "trace" a transfer lifecycle with a given ID

i.e. Transfer Request received, Initiated, Cancellation, Resumption, Completion, Failure, what errors it ran into, timestamp of completion etc  as it has historically been hard to debug.